modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -599,22 +599,29 @@
   (fn []
     (.getString rs i)))
 
+(defn- offset-date-time-value
+  "Returns the `OffsetDateTime` value of the `TIMESTAMPTZ` or `TIMESTAMPLTZ` object `timestamp`."
+  [^ResultSet rs ^Datum timestamp]
+  ;; Oracle `TIMESTAMPTZ` or `TIMESTAMPLTZ` types can have either a zone offset *or* a zone ID; you could fetch either
+  ;; `OffsetDateTime` or `ZonedDateTime` using `.getObject`, but fetching the wrong type will result in an Exception,
+  ;; meaning we have try both and wrap the first in a try-catch. As far as I know there's no way to tell whether the
+  ;; value has a zone offset or ID without first fetching a `TIMESTAMPTZ` object. So to avoid the try-catch we can
+  ;; fetch the `TIMESTAMPTZ` or `TIMESTAMPLTZ` and use `.timestampValue` instead.
+  (let [^C3P0ProxyConnection proxy-conn (.. rs getStatement getConnection)
+        conn                            (.unwrap proxy-conn OracleConnection)]
+    (t/offset-date-time (.timestampValue timestamp conn) (t/zone-offset 0))))
+
 (defmethod sql-jdbc.execute/read-column-thunk [:oracle OracleTypes/TIMESTAMPTZ]
   [_driver ^ResultSet rs _rsmeta ^Integer i]
-  ;; Oracle `TIMESTAMPTZ` types can have either a zone offset *or* a zone ID; you could fetch either `OffsetDateTime`
-  ;; or `ZonedDateTime` using `.getObject`, but fetching the wrong type will result in an Exception, meaning we have
-  ;; try both and wrap the first in a try-catch. As far as I know there's now way to tell whether the value has a zone
-  ;; offset or ID without first fetching a `TIMESTAMPTZ` object. So to avoid the try-catch we can fetch the
-  ;; `TIMESTAMPTZ` and use `.offsetDateTimeValue` instead.
   (fn []
     (when-let [^TIMESTAMPTZ t (.getObject rs i TIMESTAMPTZ)]
-      (let [^C3P0ProxyConnection proxy-conn (.. rs getStatement getConnection)
-            conn                            (.unwrap proxy-conn OracleConnection)]
-        ;; TIMEZONE FIXME - we need to warn if the Oracle JDBC driver is `ojdbc7.jar`, which probably won't have this
-        ;; method
-        ;;
-        ;; I think we can call `(oracle.jdbc.OracleDriver/getJDBCVersion)` and check whether it returns 4.2+
-        (.offsetDateTimeValue t conn)))))
+      (offset-date-time-value rs t))))
+
+(defmethod sql-jdbc.execute/read-column-thunk [:oracle OracleTypes/TIMESTAMPLTZ]
```

Comment:
Without this, the unformatted value from a query that selects a `TIMESTAMP WITH LOCAL TIME ZONE` would have type `oracle.sql.TIMESTAMPLTZ`. Now it will have a type of `java.sql.OffsetDateTime` as it would with postgres and mysql.